src/scripts/ci/travis/build.sh

```diff
@@ -55,19 +49,50 @@ fi
 # build
 if [ "${TARGETOS:0:3}" = "ios" ]; then
     xcrun --sdk iphoneos make -j 2
+elif [ "$BUILD_MODE" = "sonarqube" ]; then
+    ./build-wrapper-linux-x86/build-wrapper-linux-x86-64 --out-dir bw-outputs make -j 2
 else
     make -j 2
 fi
 
-if [ "$MODULES" != "min" ] && [ "${TARGETOS:0:3}" != "ios" ]; then
+# Run SonarQube analysis
+if [ "$TRAVIS_BRANCH" = "master" ] && [ "$TRAVIS_PULL_REQUEST" = "false" ] && [ "$BUILD_MODE" = "sonarqube" ]; then
+    # => This will run a full analysis of the project and push results to the SonarQube server.
+    #
+    # Analysis is done only on master so that build of branches don't push analyses to the same project and therefore "pollute" the results
+    echo "Starting analysis by SonarQube..."
+    sonar-scanner -Dsonar.login=$SONAR_TOKEN
+elif [ "$TRAVIS_PULL_REQUEST" != "false" ] && [ -n "${GITHUB_TOKEN-}" ]  && [ "$BUILD_MODE" = "sonarqube" ]; then
+    # => This will analyse the PR and display found issues as comments in the PR, but it won't push results to the SonarQube server
+    #
+    # For security reasons environment variables are not available on the pull requests
+    # coming from outside repositories
+    # http://docs.travis-ci.com/user/pull-requests/#Security-Restrictions-when-testing-Pull-Requests
+    # That's why the analysis does not need to be executed if the variable GITHUB_TOKEN is not defined.
+    echo "Starting Pull Request analysis by SonarQube..."
+    sonar-scanner -Dsonar.login=$SONAR_TOKEN \
+    -Dsonar.analysis.mode=preview \
+    -Dsonar.github.oauth=$GITHUB_TOKEN \
+    -Dsonar.github.repository=$TRAVIS_REPO_SLUG \
+    -Dsonar.github.pullRequest=$TRAVIS_PULL_REQUEST
+fi
+# When neither on master branch nor on a non-external pull request => nothing to do
+
+# just a test
+if [ "$BUILD_MODE" = "sonarqube" ]; then
+    echo "Starting analysis by SonarQube..."
+    sonar-scanner -Dsonar.login=$SONAR_TOKEN
+fi
+
+if [ "$MODULES" != "min" ] && [ "${TARGETOS:0:3}" != "ios" ] && [ "$BUILD_MODE" != "sonarqube" ]; then
     ./botan-test
 fi
 
 if [ "$MODULES" != "min" ] && [ "$BUILD_MODE" = "shared" ] && [ "$TARGETOS" = "native" ]
 then
-    python2 --version
+    python --version
```

Comment:
Does Ubuntu 14.04 not have a `python2` alias? If so please put a comment to that effect just above these lines, so someone doesn't come along and 'fix' it to python2 later. Using `python` for Python2 in this script is fine since we know exactly what systems Travis uses.